### DISCUSSION TITLE
How to Render DirectX view port in ImGui Window inside Dockspace

### DISCUSSION THREAD

**Discussion by sparkskapil** [ocornut/imgui#4674] [New Users Build/Link/Run issues ONLY!]
I am trying to make an engine with DirectX 9 and ImGui. 
Can anyone help me how to create a viewport window for my scene with imgui dockspace.

**Comment by rokups** [comment]
Render your viewport into a texture and render your texture using `ImGui::Image()`.

**Reply by sparkskapil** [reply]
@rokups Thanks for the reply. I have seen some tutorials to do this with OpenGL.
I am new to rendering, can you suggest some resources to implement this with DirectX9.
I have tried searching but couldn't find any tutorial or blog/post.

**Reply by meshula** [reply]
The google search you are looking for is "directx9 render offscreen surface". The primary API is this one: https://docs.microsoft.com/en-us/windows/win32/direct3d9/id3dxrendertosurface

**Comment by ocornut** [comment]
Maybe we should amend this page
https://github.com/ocornut/imgui/wiki/Image-Loading-and-Displaying-Examples

With some basic keyboard/references for each graphics API ?
